### DISCUSSION TITLE
How to disable upload / is data being sent back to openai servers ?

### DISCUSSION THREAD

**Discussion by thisintheway** [openai/whisper#1100] [Q&A] [answered]
Hi there, 

I've been trying some things with the whisper python library. I had some help from ChatGPT since i'm not super fluent in coding. It suggested using a parameter within the transcribe() function that disabled uploading data back to open ai. Here is how i set up the function : 

`result = model.transcribe(audio_data, language="fr", fp16=False, disable_upload=True)`

Which returns the error : 

`TypeError: __init__() got an unexpected keyword argument 'disable_upload'"`

That's when I realized that chatgpt suggested this based off of an old 2021 non-public version of Whisper, which does not apply to the current public version. 
Here was chatgpt's answer :
> Regarding the privacy of the data, you can prevent any audio or text data from being sent back to OpenAI by using the disable_upload parameter. This will disable the upload of data to OpenAI servers, ensuring that all processing is done locally on your machine.

As I understand it, an internet connection is required to load the model each time, at the beginning of the script. 
Here is my question: **Was this parameter removed? Does the transcribing happen locally without any upload or does it send audio and/or text data back to Openai servers?** If not, is there a way to get around this by disabling internet access to the script for a certain time ? 

This is important to me because of the confidentiality of the data used. 
Apologies if the question seems silly, I am not used to using this library.

**Comment by ItakeLs** [comment]
Using whisper locally does not send data to their servers and in the current public version there is no argument ```disable_upload```, everything stays local. As for their new API, I am not sure if data is sent to their servers and you would probably have to check their policies.

**Comment by jongwook** [comment, marked as answer]
ChatGPT often generate factually inaccurate sentences, and this codebase did not exist in 2021.

You can rest assured that your audio data is not being uploaded to anywhere. Once the model weights are downloaded, this whisper package will run without an internet connection.
